classes/mail/Mail.inc.php

```diff
@@ -362,11 +362,24 @@ function getReplyToString($send = false) {
 	}
 
 	/**
-	 * Set the subject of the message.
+	* Set the subject of the message.
+	* @param $subject string
+	*/
+	function setSubject($subject) {
```

Comment:
`setSubject` is just a setter function, so I'd rather we kept anything more complicated than that outside of it.

We have a convention to avoid XSS flaws in emails that looks a little like this:
- In email templates, you can use replacement variables like `{$contextName}` so that the templates are personalized before sending.
- Just about all of those variables appear in the email body -- but a few do appear in the subject, like in this case. The same variables and values are available to both places. (You can also look at the Announcements tool for another example where this problem shows up; if you put a `&` in the announcement title it'll also come out as `&amp;` when sent.)
- Email bodies are in HTML format, and XSS escaping is expected to be done when assigning variables to the template, which is when we know how to escape them (based on whether they are HTML or plaintext values, for example). [Example](https://github.com/pkp/pkp-lib/blob/87db0786c680408d0dab0ccc56e6e750f159e748/classes/mail/MailTemplate.inc.php#L151)
- So you can see we have a mismatch: variables are provided to the template in HTML format, and then applied to both the body and the subject as required. However, subjects expect plaintext, and bodies expect HTML.
- You can see where variables are replaced in both the body and the subject in the [Mail::send](https://github.com/pkp/pkp-lib/blob/87db0786c680408d0dab0ccc56e6e750f159e748/classes/mail/MailTemplate.inc.php#L222) function (for `stable-3_3_0`; this has changed in `stable-3_4_0` and `main`). Here, the same approach is applied to both body and subject -- I think that's the heart of the bug.

So I'd suggest fixing the escaping of the subject in the `MailTemplate::replaceParams` function.